sources/file-system/src/main/java/io/smallrye/config/source/file/FileSystemConfigSource.java

```diff
@@ -77,12 +77,13 @@ public FileSystemConfigSource(File dir, int ordinal) {
 
     private static Map<String, String> scan(File directory) {
         if (directory != null && directory.isDirectory()) {
-            try (Stream<Path> stream = Files.walk(directory.toPath())) {
+            try (Stream<Path> stream = Files.walk(directory.toPath(), 1)) {
```

Comment:
there is also a variant to follow symlinks. Should we use it?

java.nio.file.Files#walk(java.nio.file.Path, int, java.nio.file.FileVisitOption...)

Reply:
I don't think we need to follow symlink, we don't want to enter sub directory. if a file is a symlink, then it will be opened. Perhaps I am missing advanced case with symlinks?